specification/sql/resource-manager/Microsoft.Sql/2017-03-01-preview/managedDatabases.json

```diff
@@ -0,0 +1,556 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2017-03-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Sql/locations/{locationName}/managedDatabaseRestoreAzureAsyncOperation/{operationId}/completeRestore": {
```

Comment:
What is the intent of this API?

Reply:
It completes a restore that we previously started (by creating a database with parameter createMode : externalBackup).

Reply:
How does a user determine the operation id that they need to restore, after the create is completed? And why does a user need to manually complete instead of it completing itself?

Reply:
When a user creates a database with create mode externalBackupRestore, we give him a request id, and a location (url) where he or she  can check for status of this operation. 
Log replay is a minimum downtime migration scenario. We want to restore backups as we are backing them up to remote storage, so the customer can stay operational during this time, and catch up the last tiny bit of difference only when the customer is ready to go to read-only mode for not more than 5 minutes. Then we issue the complete restore request with a log backup file that's been made after we went to read only mode.
So the answer to your second question is: We generally don't have the last backup that's used to complete restore when we start the migration.